app/assets/scripts/components/upload-form.js

```diff
@@ -62,9 +62,8 @@ var UploadForm = React.createClass({
           failures = ['No data provided'];
           this.logOutput(failures);
         }
-        // Check header on line 1
+        // Check header on first line
         if (line === 0) this.checkHeader(data, failures);
```

Comment:
the function `checkHeader` is still accepting an array as a parameter which is being modified internally, instead of returning a new array of failures.

Also the `checkHeader` should do just that, check the header for errors. `this.logOutput(failures);` should be done outside that function.

Perhaps something like:
```
if (line === 0) {
  failures = failures.concat(this.checkHeader(data));
}
if (failures.length) {
  this.logOutput(failures);
}
```